### PR TITLE
Store restored channel immediately

### PR DESCRIPTION
**Pull request by pm47**
Builds on #448.

When we restore a channel from a peer backup, we need to store the channel in database as soon as possible, so we are exactly in the same state as if the channel was restored from local db.

This applies whether we are restoring from empty or outdated local db.

This is the backport of 626deed7e2c9e207f085b3e0b53c9e78e42d45e2 on master.

### CODE REVIEW

File: src/commonMain/kotlin/fr/acinq/lightning/channel/Syncing.kt [hunk 1]
```diff
@@ -49,7 +46,13 @@ data class Syncing(val state: PersistedChannelState, val waitForTheirReestablish
                         state
                     }
                     val channelReestablish = nextState.run { createChannelReestablish() }
-                    val actions = listOf(ChannelAction.Message.Send(channelReestablish))
+                    val actions = buildList {
+                        if (nextState != state) {
+                            // we just restored from backup
+                            add(ChannelAction.Storage.StoreState(nextState))
```
Comment (t-bast) [inline src/commonMain/kotlin/fr/acinq/lightning/channel/Syncing.kt:52]: We can probably update one of our restore unit tests to check that `StoreState` is emitted here?

Comment (pm47) [reply src/commonMain/kotlin/fr/acinq/lightning/channel/Syncing.kt:52]: Clearly, I'm in-between rebases and this was meant as a reminder. Let put it back to Draft.

Comment (pm47) [reply src/commonMain/kotlin/fr/acinq/lightning/channel/Syncing.kt:53]: Done in [8b37758](https://github.com/ACINQ/lightning-kmp/pull/429/commits/8b377580ebbdf362b2a285b63be7a172d2267668).

**Comment by pm47** [comment]
Rebased on #448.

**Comment by pm47** [comment]
Superseded by #454.

**Pull request closed without merging**